
%%%%%%%%%%%%%%%%%%%%%%% file typeinst.tex %%%%%%%%%%%%%%%%%%%%%%%%%
%
% This is the LaTeX source for the instructions to authors using
% the LaTeX document class 'llncs.cls' for contributions to
% the Lecture Notes in Computer Sciences series.
% http://www.springer.com/lncs       Springer Heidelberg 2006/05/04
%
% It may be used as a template for your own input - copy it
% to a new file with a new name and use it as the basis
% for your article.
%
% NB: the document class 'llncs' has its own and detailed documentation, see
% ftp://ftp.springer.de/data/pubftp/pub/tex/latex/llncs/latex2e/llncsdoc.pdf
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\documentclass[runningheads,a4paper]{llncs}

\usepackage{amssymb}
\setcounter{tocdepth}{3}
\usepackage{graphicx}
\usepackage{cite}
\usepackage{url}
\usepackage{natbib}
\urldef{\mailsa}\path|{samuel.neaves}@kcl.ac.uk|


\renewcommand{\bibname}{\leftline{References}}
   
\newcommand{\keywords}[1]{\par\addvspace\baselineskip
\noindent\keywordname\enspace\ignorespaces#1}

\begin{document}

\mainmatter  % start of an individual contribution


\title{ILP Identifies Pathway Activation Patterns}
% oe Identifying Pathway Activation Patterns- For Classification
%ILP identifies pathway activation patterns. 
%n Application of ILP to Systems Biology
% a short form should be given in case it is too long for the running head
\titlerunning{Lecture Notes in Computer Science: Authors' Instructions}

% the name(s) of the author(s) follow(s) next
%
% NB: Chinese authors should write their first names(s) in front of
% their surnames. This ensures that the names appear correctly in
% the running heads and the author index.
%
\author{Samuel R Neaves}
%
\authorrunning{Lecture Notes in Computer Science: Authors' Instructions}
% (feature abused for this document to repeat the title also on left hand pages)

% the affiliations are given next; don't give your e-mail address
% unless you accept that it will be published
\institute{Deparment of Informatics King's College London,\\
Strand, London, UK\\
\mailsa\\
%\mailsc\\
\url{http://www.kcl.ac.uk}}

%
% NB: a more complex sample for affiliations and the mapping to the
% corresponding authors can be found in the file "llncs.dem"
% (search for the string "\mainmatter" where a contribution starts).
% "llncs.dem" accompanies the document class "llncs.cls".
%

\toctitle{Lecture Notes in Computer Science}
\tocauthor{Authors' Instructions}
\maketitle


\begin{abstract}
We show how a logical aggregation method combined 
with propositionalization methods can construct novel structured biological features for 	classification and ranking tasks in systems biology. 

\keywords{Biological Pathways, Reactome, RDF, Barcode, Logical Aggregation, Warmr, Treeliker, ILP.}
\end{abstract}



\section{Introduction and Background}

%This paper describes a method to use Inductive Logic Programming(ILP) techniques to mine structured biological data. 
%Overview
In the field of Systems Biology researchers are often interested in identifying perturbations within a biological system that are different across experimental conditions. In this paper we use the example of identifying these perturbations in two types of Lung Cancer (for details  see \citep{rhrissorrakrai_sbv_2013}).
To clarify we are looking to identify differences rather than building a diagnostic tool for lung cancer.

% typical approach
A typical pipe line for this kind of task has three distinct stages. The first stage is to use a technology such as a microarray or RNAseq to measure gene expression across the genome in a number of samples from each of the experimental conditions. 
The second stage is to identify a subset of genes whose expression values correlate with the conditions.
Stage 2 is commonly achieved by performing differential expression analysis and ranking genes by their fold change values. A statistical test is then used as a cut off value to identify the relevant set to take forward to stage 3. 
Alternatively for stage 2 researchers may train a model using machine learning to classify samples into experimental conditions, often using an attribute value representation where features are a vector of gene expression values. 
This approach has the advantage that the constructed model may have found dependencies between genes which would not have been identified otherwise.
%(experimental conditions). 
Researchers will use the 'top' features from the model to identify the set of genes to take on to stage 3. 
%Genes that are identified in this manner may be  different to a simple list of differently expressed genes as the constructed model , which are not highly differently expressed. 


%We note classification is often not the true task -the classification performance is used as a measure of the quality of the identified differences. Classification might not be the true task as the class of a sample may already be known - for example where the biopsy is taken from may indicate the class. 
%typical approach stage 2
In stage 3  researchers look for connections between these genes, for example by performing Gene Set Enrichment Analysis(GSEA) \citep{subramanian_gene_2005}. Here the set of identified genes are compared with predefined sets of genes. The predefined sets of genes indicate a known relation. For example having a related function, existing in the same location in the cell or taking part in a pathway. A further way a set could be defined is a topological set, this takes into account finer grained internal relations to define a set. For example a topological set could be identified by community detection in a gene regulatory network.

%related relational work
In contrast to the two stage approach, analysis can be performed in one step. Past ILP research \citep{gamberger_induction_2004} has integrated the two steps of finding differently expressed genes and GSEA by using relational subgroup discovery. These have used the hierarchical Gene Ontology to relate genes and has the advantage of being able to construct novel sets by sharing variables across predicates that define the sets. For example a set could be defined as the genes that have been annotated with two GO terms.  Other ways researchers have tried to use known relations is by adapting the classification approach. New features are built by aggregating across a predefined set of genes - for example by taking an average expression value for a pathway, see ~\citep{holec_comparative_2012} for a review of these methods.

In most of these methods it is common to ignore the detailed relations between entities in a pathway, the pathway is treated as an unstructured collection of genes. Only do topologically defined sets take advantage of any known internal relations. But often these create crude clusters of genes, and do not follow biological intuition of information flow through the pathway.In addition the chosen network representation is often inappropriate for pathway analysis as described later . Therefore a major limitation of the current classification approaches is that the models constructed are limited to being constructed from either genes or crude aggregates of sets of genes. However biological researchers are not only interested in these kind of models. More general pathway activation patterns are also of interest, including paths and loops where some of the entities could possibly be represented by uninstantiated variables. Two examples are 1. Consistency modelling and 2. Network motif finding. In consistency modelling \citep{guziolowski2010analysis} a pattern of inconsistency is matched against gene regulatory networks and for each match Answer Set Programming techniques are used to amend the pathways to remove the inconsistency. Similar work has been done in ILP  where biological pathways have been constructed and amended \citep{ray2010automatic}. In network motif finding \citep{kim_biological_2011}, again the biological intuition of information flow is often ignored and it is uncommon to either take into account the class of the network sample or to include uninstantiated variables in the search for motif patterns. Motifs are most commonly searched for across a network constructed from an agglomerative of the experimental data rather than by mapping expression values to a standard graph of the pathway which would in effect give an instantiated graph for each sample. This approach would allow for the searching of common pathway activation patterns across samples - taking into account the class of each sample. 

In contrast to the work on consistency modelling, in this work we assume the pathways are in some sense correct and we are looking at individuals sample expression patterns mapped on to the pathway. This work is most similar to  \citep{holec2008using} in which the authors propose using Fully Coupled Fluxes as features. An FCF is the longest possible chain of vertices in which non-zero vertex activation implies a certain (non-zero) activation in its successors.

%\section{Aim}
%

	  
\section{Method}
The aim of this paper is to identify pathway activation patterns that differ between biological samples of different classes, in order to give a biologist different information than models built from simple gene features. We use structural pathway information to build first order features that are used to construct classification and ranking models, which  provide an appropriate level of abstraction for the interpretation of the model to be insightful for a biologist. 
\subsection{Raw Data}
The GEO data set we use for this study is GSE2109. This is a two class data set with the classes corresponding to different types of lung cancer(SCC and AC).There are 38 SCC samples and 33 AC samples. .
	
Reactome \citep{croft_reactome_2013} is a collection of manually curated peer reviewed pathways. Reactome is made available as an RDF XML Biopax level 3 file. This allows for simple passing using SWI-Prolog's semantic web libraries. 
In this work we use a high level logical abstraction of the idea of a biological pathway. In biological research it is common to see many types of biological networks represented by mathematical graphs. When considering biological pathways simple directed networks of genes and proteins are not appropriate, because this formalism does not adequately model the dependencies of activation patterns- either bipartite graphs or hypergraphs are required see \citep{whelan2011representation} for more details on this. We address this problem by using a directed reaction centric graph which we have extracted from Reactome.
\subsection{Data Processing}
Boolean networks~\citep{wang_boolean_2012} are a common abstraction in biological research, but these are normally applied at the gene or protein level not at the reaction level. In order to use a boolean network abstraction on a reaction network, we apply a logical aggregation method from the measured probes in the microarray to reactions as defined in Reactome.

We first discretize the probe values into binary values using Barcode \citep{mccall_gene_2014} which is a tool for applying learnt thresholds to microarray data,  Barcode makes it possible to compare gene expressions, both within a sample and between samples potentially measured by different arrays. 

Once we have binary probe values, we use the structure provided by
the Reactome RDF graphs and our biological understanding to build Reaction level features.
Reactome defines reactions with entities divided into substrates, enzymes and products. Each reaction has a set of inputs and set of outputs, in addition a reaction may be controlled (activated or inhibited) by an entity. Entities in Reactome include protein complexes and protein sets, A protein complex or protein set may itself include sub complexes or sub sets. This gives the input to a reaction a kind of tree structure. For the logical aggregation step we interpret this structure as a simple logical circuit. The relationship between probes and proteins is treated as an OR gate, protein complexes as and AND gate, and protein sets also as an OR gate. With a final step taking into account the activation or inhibition of the reaction by any controlling entities. In this way we can say that a reaction 'can proceed' if and only if, the input circuit evaluates to true. If a reaction 'can proceed' then we say that it is 'on' otherwise we say that it is 'off'. 

\subsection{Searching for Pathway Activation Patterns}
In this work we have experimented with two propositionalization methods, separately and then combining them. We search for features independently in each pathway and with a language bias that suitably constrains the search in order to achieve a manageable number of structures. 

The first method Warmr \citep{dehaspe_mining_1997} is the first order equivalent of item set and association rule mining.It can be used as propositionalization method by independently searching for frequent queries in the two classes. An advantage of using Warmr is that it is easy to define background predicates for relevant concepts. For example a path or loop of all 'on' reactions. As Warmr does not prune by relevance to classification tasks it can however quickly build to an intractable search with many irrelevant or similar queries/features built. 
	     

The second method, Tree liker~\citep{vzelezny2013fast} is a modern ILP tool that implements a number of algorithms, it has been shown to produce long features by building features bottom up in blockwise manner. This is a desirable trait for  this task as longer features will give more information to a biologist. A limitation is that the features are 'tree like' which means there can be no cycles in the variables. Treeliker unlike Warmr does not support explicit background knowledge and therefore all relevant relations need to be preprocessed using prolog.

For a combined method we first take a top feature constructed by Treeliker and use this as the basis for the language bias input into Warmr. We then add further language bias constraints that guide Warmr into searching for frequent queries that create cycles in the Treelike features. This results in longer features that contain cycles than Warmr can find on its own. 
	     
	     
\section{Results}
As we are interested in producing comprehensible models, we limit our experiments to rule and tree building algorithms.We use Wekas implementation of classifications trees (J48) and rule learning (JRip). The best classifiers achieve 80 \% accuracy in 10 fold cross validation.
(Best rankers)
\newline

\noindent
{\it{Example Warmr found feature}}
\begin{verbatim}
microarray(A),reaction(A,B,1),reaction(A,C,0),link(C,B,D),link(B,C,E).
\end{verbatim}
\noindent 
This is a simple cyclical feature, the variable A matches one sample. 
\newline

\noindent
{\it{Example Tree Liker found feature}}

\begin{verbatim}
reaction(A, 0), link(A, B, follows), reaction(B, 1), link(B, C, _),    
reaction(C, 0), link(A, D, activation), reaction(D, 0).
\end{verbatim}
\noindent 
This is a longer tree liker feature -notice Treeliker does not require a variable for the sample.  
\newline

\noindent
{\it{Example combined method found feature}}
\begin{verbatim}
microarray(A),reaction(A,B,0),link(B,C,follows),reaction(A,C,1),
link(C,D,E),reaction(A,D,0),link(B,F,activation),reaction(A,F,1),
link(F,D,E),link(D,G,E),reaction(A,G,0)
\end{verbatim}
\noindent
This final feature is both long and contains a cycle. With the structured features found, we were able to achieve a comparable 'accuracy' to that of a model built with raw expression values - but now we have identified Pathway activation perturbations rather than just gene expression perturbations. 

\section{Discussion}
This work has shown the appropriateness of using ILP methods to mine the abundance of highly structured biological data.Using this method we have identified differences in pathway activation patterns that go beyond the standard analysis of differentially expressed genes, enrichment analysis, gene feature ranking and pattern mining for common network motifs. We have also demonstrated the use of logical aggregation to a  reaction graph and how this simplifies the search for hypothesises to an extent where searching all pathways is tractable. The extension of Treeliker found features with Warmr to connect cycles is believed to be novel. This enables the finding of long cyclical features which are of interest to biologists. 
\newline



%\subsubsection*{Acknowledgments.} Thanks to ...


\begingroup
\let\clearpage\relax
\bibliography{mybib}{}
\endgroup

%\bibliography{mybib}{}
%\bibliographystyle{plain}
\bibliographystyle{unsrt}
%\bibliographystyle{unsrtnat}
%\bibliographystyle{ieeetr}





\end{document}
